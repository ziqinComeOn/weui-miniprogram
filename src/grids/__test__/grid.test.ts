import path from 'path'
import simulate from 'miniprogram-simulate'

describe('grids', () => {
    const grids = simulate.load(path.resolve(__dirname, '../grids'), 'mp-grids')

    const id = simulate.load({
        template: `
            <mp-grids grids="{{grids}}"></mp-grids>
        `,
        usingComponents: {
            'mp-grids': grids
        },
        data: {
            grids: Array(9)
                .fill(null)
                .map((_, i) => ({
                    imgUrl: `/img-${i}.png`,
                    url: `/${i}`,
                    text: `Grid${i}`
                }))
        }
    })

    test('basic', async () => {
        const comp = simulate.render(id)
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })
})
