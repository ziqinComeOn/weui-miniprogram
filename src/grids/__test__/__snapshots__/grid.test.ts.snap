// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grids basic 1`] = `
<main>
  <mp-grids>
    <wx-view
      class="weui-grids "
    >
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/0"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-0.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid0
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/1"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-1.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid1
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/2"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-2.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid2
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/3"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-3.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid3
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/4"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-4.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid4
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/5"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-5.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid5
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/6"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-6.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid6
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/7"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-7.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid7
        </wx-view>
      </wx-navigator>
      <wx-navigator
        appId=""
        class="weui-grid"
        extraData=""
        hoverClass="navigator-hover"
        hoverStartTime="{{50}}"
        hoverStayTime="{{600}}"
        hoverStopPropagation="{{false}}"
        openType="navigate"
        path=""
        target="self"
        url="/8"
        version="release"
        bind:complete=""
        bind:fail=""
        bind:success=""
      >
        <wx-view
          class="weui-grid__icon"
        >
          <wx-image
            alt="{{true}}"
            class="weui-grid__icon_img"
            src="/img-8.png"
          />
        </wx-view>
        <wx-view
          class="weui-grid__label"
        >
          Grid8
        </wx-view>
      </wx-navigator>
    </wx-view>
  </mp-grids>
</main>
`;
